components/admin_console/plugin_management/plugin_management.jsx

```diff
@@ -967,8 +989,8 @@ export default class PluginManagement extends AdminSettings {
                             <div className='col-sm-8'>
                                 <div className='file__upload'>
                                     <button
-                                        className={uploadBtnClass}
-                                        disabled={!enableUploads || !enable}
+                                        className={classNames(['btn', {'btn-primary': enableUploads}])}
+                                        disabled={!enableUploads || !this.state.enable || this.state.requirePluginSignature}
```

Comment:
Is `enable` part of the state? I believe it's actually using the props variable from [here](https://github.com/mattermost/mattermost-webapp/blob/5bdd64f1e63f22d923dbf0c88e581c8503b5a153/components/admin_console/plugin_management/plugin_management.jsx#L798): 

`const enable = this.props.config.PluginSettings.Enable;`

Reply:
@marianunez 
The problem with using the `enable` variable is that it's a prop and not what's stored locally. Which means that if I toggled the `Enabled Plugin` option in the UI it wouldn't update the `Upload` button. However, if I saved it then it would. I believe the problem exists as is, without my change. So this kinda fixes that.

Reply:
But where is it stored locally though?

Reply:
In the component state, which is triggered by this [function](https://github.com/mattermost/mattermost-webapp/blob/2425f3da40c2ca6aa7f445637e3f77c618e6fa4a/components/admin_console/plugin_management/plugin_management.jsx#L976), which then updates the state [here](https://github.com/mattermost/mattermost-webapp/blob/afa982424965523c479b1a0eaf921e6f826e7720/components/admin_console/admin_settings.jsx#L60). Since this plugin setting component is a child of [AdminSetting](https://github.com/mattermost/mattermost-webapp/blob/2425f3da40c2ca6aa7f445637e3f77c618e6fa4a/components/admin_console/plugin_management/plugin_management.jsx#L397) it updates the state for us.

Reply:
Ah a child of `AdminSetting`, gotcha. That's what I was missing.

Reply:
Yeaaa.. Inhertience 😭 
https://twitter.com/dan_abramov/status/752643494972383232/photo/1
![image](https://user-images.githubusercontent.com/25732808/68056371-55d29180-fcc9-11e9-9fa3-bdd4e232bd8f.png)
https://programmingwithmosh.com/react/react-composition-vs-inheritance/